src/object/cli_obj.c

```diff
@@ -3385,18 +3385,25 @@ obj_shard_list_comp_cb(struct shard_auxi_args *shard_auxi,
 
 	shard_arg = container_of(shard_auxi, struct shard_list_args, la_auxi);
 	if (obj_auxi->req_tgts.ort_grp_size == 1) {
-		if (obj_auxi->opc == DAOS_OBJ_RECX_RPC_ENUMERATE &&
-		    shard_arg->la_recxs != NULL) {
+		if (obj_auxi->opc == DAOS_OBJ_RECX_RPC_ENUMERATE ||
+		    shard_arg->la_recxs == NULL) {
```

Comment:
@liuxuezhao , I have merged the last modification from #10531 : Thank you very much to have pointing me this 👍 

I am waiting for the CI validation as I am not understanding this modification which should lead to a segfault when the function `daos_obj_list_recx()` is called with its parameter recxs equal to `NULL`. For example, in the _daos-serialize_ code, it is used to retrieve the size of the record.

Reply:
need confirm with @wangdi1 , seems should be 
"if (obj_auxi->opc == DAOS_OBJ_RECX_RPC_ENUMERATE &&
		    shard_arg->la_recxs != NULL)"?